physlr/physlr.py

```diff
@@ -1750,23 +1750,16 @@ def index_minimizers_in_backbones(backbones, bxtomxs):
         mxtopos = {}
         for tid, path in enumerate(progress(backbones)):
             for pos, u in enumerate(path):
-                if u not in bxtomxs:
-                    u = u.rsplit("_", 1)[0]
-                if u not in bxtomxs:
-                    u = u.rsplit("_", 1)[0]
-                for mx in bxtomxs[u]:
-                    mxtopos.setdefault(mx, set()).add((tid, pos))
-        print(
-            int(timeit.default_timer() - t0),
-            "Indexed", len(mxtopos), "minimizers", file=sys.stderr)
-        return mxtopos
-
-    @staticmethod
-    def index_minimizers_in_backbones_split(backbones, bxtomxs):
-        "Index the positions of the minimizers in the backbones."
-        mxtopos = {}
-        for tid, path in enumerate(progress(backbones)):
-            for pos, u in enumerate(path):
+                if Physlr.args.map_type == "all":
+                    if u not in bxtomxs:
+                        u = u.rsplit("_", 1)[0]
+                    if u not in bxtomxs:
+                        u = u.rsplit("_", 1)[0]
+                elif Physlr.args.map_type == "split":
+                    pass
+                else:
+                    print("Invalid --map-type argument", file=sys.stderr)
```

Comment:
I think this parameter check should be before the for loop and the function should exit with `sys.exit` -- Otherwise you'd see this print statement over and over for each line in the file